lib/class-wp-theme-json-gutenberg.php

```diff
@@ -796,6 +819,23 @@ protected static function sanitize( $input, $valid_block_names, $valid_element_n
 		$schema['settings']           = static::VALID_SETTINGS;
 		$schema['settings']['blocks'] = $schema_settings_blocks;
 
+		// For settings.typography.fontFamilies, make the $schema have all indexes present in the $input.
+		if ( isset( $input['settings']['typography']['fontFamilies'] ) ) {
+			// Do not handle the cases where the sanitization is called before font families are merged, since it will be handled again later.
+			if ( isset( $input['settings']['typography']['fontFamilies']['theme'] ) || isset( $input['settings']['typography']['fontFamilies']['custom'] ) ) {
```

Comment:
There are some cases where `settings.typography.fontFamilies` have a "source" which is either `theme` or `custom`. I thought this was the simplest way to handle this case because it breaks the validation otherwise. But I'm not sure if it exposes an insecurity since we're returning early.

Reply:
Addressed this with @vcanales in [38263a1](https://github.com/WordPress/gutenberg/pull/53273/commits/38263a16151cc181a869bee893653b679c49ba54).